Account period totals

lomamech opened the pull request:
Build Account by Period Totals Report  …
- Update uigrid of genereal ledger, Added columns for each monthly total periods
- Create template for the balance cell for eachn period
- Adding a function who returns all period of current fiscal Year

closes #1637 .  Closes #1676.

@jniles I will wait for your fead back finally to be able to restore the old presentation ( @mbayopanda  ) of the u grid of general ledger

lomamech committed:
Build Account by Period Totals Report
- Update uigrid of genereal ledger, Added columns for each monthly total
  periods
- Create template for the balance cell for eachn period
- Adding a function who returns all period of current fiscal Year

jniles commented on client/src/modules/general-ledger/general-ledger-accounts.ctrl.js:
```diff
@@ -25,30 +25,105 @@ function GeneralLedgerAccountsController(GeneralLedger, Session, Notify,
+      enableFiltering  : false,
+      headerCellFilter : 'translate',
+      headerCellClass  : 'text-center',
+      cellTemplate     : '/modules/general-ledger/templates/balance1.cell.html' },
```
Hmm... There must be an easier way to do this.

Normally, you should use an HTML template, but all these HTML templates are identical.

Try this - it will save a lot of code/HTTP requests.
```js
function getCellTemplate(key) {
  return '<div class="ui-grid-cell-contents text-right">' +
    '<div ng-show="row.entity.' + key +'" >' +
      '{{ row.entity.' + key +' | currency: grid.appScope.enterprise.currency_id }}' +
    '</div>' + 
  '</div>';
}

// later, in your columnDefs
{ field : 'balance', cellTemplate: getCellTemplate('balance1') }
```

jniles commented on server/controllers/finance/generalLedger/index.js:
```diff
@@ -69,20 +78,41 @@ function listAccounts(req, res, next) {
+        , periode${period.id}.balance AS balance${period.number}
+      `;
+
+      jointure += `
```
I don't understand this code ... why do we need to LEFT JOIN `general_ledger` on `period_total`?  Why can't we just pull all the data from `period_total`?

lomamech committed:
Improve code and request
- Remove templates for each balance by period and using a function for get
  it
- Update function get all periods of current fiscal year, selected also
  period Zero
- Improve request SQL remove Instruction LEFT JOIN by the using of CASE
  THEN ELSE
- Refactong the calcul of account balance by the using of table
  period_total instead the table general_ledger

jniles commented on server/controllers/finance/reports/reportAccounts/index.js:
```diff
@@ -119,7 +119,7 @@ function getAccountTransactions(accountId, source, dateFrom, dateTo, openingBala
         WHERE account_id = ? ${dateCondition}
         GROUP BY record_uuid
         ORDER BY trans_date ASC
-      )c, (SELECT @cumsum := ${openingBalance})z
```
:+1:

lomamech committed:
Build Account by Period totals Report
- Update general ledger ui grid adding column balance, add balance of
  period totals for each month of the fiscal year
- Update request for general ledger api adding the jointure on table
  period total
